src/ui/header/Header.tsx

```diff
@@ -3,16 +3,26 @@ import Box from '@mui/material/Box';
 import Toolbar from '@mui/material/Toolbar';
 import Typography from '@mui/material/Typography';
 import Button from '@mui/material/Button';
+import { useNavigate } from 'react-router-dom';
 
 import { useLocale } from 'hooks';
 import { HeaderProps } from './Header.types';
 import { LanguageSwitch } from './languageSwitch/LanguageSwitch';
 import { styles, StyledLink } from './Header.styles';
 import { AppRoute } from 'routing/AppRoute.enum';
 
-export const Header = ({ isLoggedIn }: HeaderProps) => {
+export const Header = ({ isLoggedIn, onLogoutClick }: HeaderProps) => {
   const { locale, setLocale, formatMessage } = useLocale();
 
+  const navigate = useNavigate();
+
+  const handleButtonClick = () => {
+    if (!isLoggedIn) {
+      navigate(AppRoute.home);
+    } else onLogoutClick();
+    navigate(AppRoute.login);
```

Comment:
Hmm... jak nie jest zalogowany, to robi dwa navigate, dobrze widzę? (pierwszy nie terminuje funkcji)

Reply:
Heh, jakoś to w ogóle dziwnie napisałem xD